ui/desktop/src/components/settings_v2/extensions/agent-api.ts

```diff
@@ -0,0 +1,184 @@
+
+/**
+ * Makes an API call to the extension endpoints
+ */
+export async function extensionApiCall(
+  endpoint: string,
+  payload: any,
+  options: ToastServiceOptions = {}
+): Promise<Response> {
+  // Configure toast notifications
+  toastService.configure(options);
+
+  // Determine if we're activating or removing an extension
+  const isActivating = endpoint == '/extensions/add';
+  const action = {
+    type: isActivating ? 'activating' : 'removing',
+    verb: isActivating ? 'Activating' : 'Removing',
+    pastTense: isActivating ? 'activated' : 'removed',
+  };
+
+  // for adding the payload is an extensionConfig, for removing payload is just the name
+  const extensionName = isActivating ? payload.name : payload;
+  let toastId;
+
+  // Step 1: Show loading toast (only for activation)
+  if (isActivating) {
+    toastId = toastService.loading({
+      title: extensionName,
+      msg: `${action.verb} ${extensionName} extension...`,
+    });
+  }
+
+  try {
+    // Step 2: Make the API call
+    const response = await fetch(getApiUrl(endpoint), {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Secret-Key': getSecretKey(),
+      },
+      body: JSON.stringify(payload),
+    });
+
+    // Step 3: Handle non-successful responses
+    if (!response.ok) {
+      return handleErrorResponse(response, extensionName, action, toastId);
+    }
+
+    // Step 4: Parse response data
+    const data = await parseResponseData(response);
+
+    // Step 5: Check for errors in the response data
+    if (data.error) {
+      const errorMessage = `Error ${action.type} extension: ${data.message || 'Unknown error'}`;
+      toastService.dismiss(toastId);
+      toastService.error({
+        title: extensionName,
+        msg: errorMessage,
+        traceback: data.message || 'Unknown error',
+      });
+      throw new Error(errorMessage);
+    }
+
+    // Step 6: Success - dismiss loading toast and return
+    toastService.dismiss(toastId);
+    toastService.success({
+      title: extensionName,
+      msg: `Successfully ${action.pastTense} extension!`,
+    });
+    return response;
+  } catch (error) {
+    // Final catch-all error handler
+    toastService.dismiss(toastId);
+    console.error(`Error in extensionApiCall for ${extensionName}:`, error);
+    throw error;
+  }
+}
+
+// Helper functions to separate concerns
+
+// Handles HTTP error responses
+function handleErrorResponse(
+  response: Response,
+  extensionName: string,
+  action: { type: string; verb: string },
+  toastId: string
+): never {
+  const errorMsg = `Server returned ${response.status}: ${response.statusText}`;
+  console.error(errorMsg);
+
+  // Special case: Agent not initialized (status 428)
+  if (response.status === 428 && action.type === 'activating') {
+    toastService.dismiss(toastId);
+    toastService.error({
+      title: extensionName,
+      msg: 'Agent is not initialized. Please initialize the agent first.',
```

Comment:
nit: What do you think about something more generic in the message here? If I put myself in the shoes of a user not familiar with the implementation of goose, I wouldn't know what "initialize the agent" means